### PR TITLE
Feat/preflight check

### PR DESCRIPTION
**Pull request by fhennig**

### CODE REVIEW

File: build_product_images.py [hunk 1]
```diff
@@ -264,6 +265,14 @@ def check_architecture_input(architecture):
     return architecture
 
 
+def get_tags_for_product(bakefile, product: str) -> List[str]:
+    tags = []
+    targets = bakefile["group"][product]["targets"]
+    for t in targets:
+        tags.extend(bakefile["target"][t]["tags"])
```
Comment (nightkr) [inline build_product_images.py:272]: `product` can actually also be a target (for example: "I only want to build Hadoop 3.3.4") or a meta-group ("I want to build all versions of all products"). I think a more correct answer would be something like:
```suggestion
    for target in bakefile["group"].get(product, {}).get("targets", []):
        tags.extend(get_tags_for_product(bakefile, target))
    tags.extend(bakefile["target"].get(product, {}).get("tags", [])
```

Comment (nightkr) [reply build_product_images.py:272]: It might also be more correct to name the function something like `get_tags_for_target`.

Comment (fhennig) [reply build_product_images.py:272]: I see!

**Comment by razvan** [comment]
Alternative PR: https://github.com/stackabletech/docker-images/pull/339

**Comment by fhennig** [comment]
superseded by the alternative PR linked above

**Pull request closed without merging**